compiler/src/iree/compiler/Codegen/LLVMGPU/VectorReductionToGPU.cpp

```diff
@@ -111,22 +139,23 @@ struct LLVMGPUReduceToGPUPass
       (void)applyPatternsAndFoldGreedily(getOperation(), std::move(patterns));
     }
 
+    auto workgroupSize = llvm::to_vector<4>(llvm::map_range(
+        getEntryPoint(funcOp)->workgroup_size().getValue(),
+        [&](Attribute attr) { return attr.cast<IntegerAttr>().getInt(); }));
+    assert(workgroupSize[1] == 1 && workgroupSize[2] == 1);
```

Comment:
What made this guaranteed? Can you add a comment?

Reply:
the KernelConfig.cpp code.

Reply:
So, I believe the meaning of workgroup size is different from the workgroup count at the flow level? At the flow level, the workgroup count shows the total number.

Reply:
workgroup size is the number of threads per workgroup. That's what we care about here. This is not the same as workgroup count or workload per work indeed.
Here we want the number of threads to know how to distribute the work.

Reply:
Thanks for the explanation. The words are so similar to each other but representing a different concept.